gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -733,6 +739,143 @@ impl ILnRpcClient for GatewayLndClient {
             failure_reason: "Gatewayd has not started to route HTLCs".to_string(),
         })
     }
+
+    async fn connect_to_peer(
+        &self,
+        pubkey: String,
+        host: String,
+    ) -> Result<EmptyResponse, LightningRpcError> {
+        let mut client = Self::connect(
+            self.address.clone(),
+            self.tls_cert.clone(),
+            self.macaroon.clone(),
+        )
+        .await?;
+
+        match client
+            .lightning()
+            .connect_peer(ConnectPeerRequest {
+                addr: Some(LightningAddress { pubkey, host }),
+                perm: false,
+                timeout: 10,
+            })
+            .await
+        {
+            Ok(_) => Ok(EmptyResponse {}),
+            Err(e) => Err(LightningRpcError::FailedToConnectToPeer {
+                failure_reason: format!("Failed to connect to peer {e:?}"),
+            }),
+        }
+    }
+
+    async fn get_funding_address(&self) -> Result<GetFundingAddressResponse, LightningRpcError> {
+        let mut client = Self::connect(
+            self.address.clone(),
+            self.tls_cert.clone(),
+            self.macaroon.clone(),
+        )
+        .await?;
+
+        match client
+            .wallet()
+            .next_addr(AddrRequest {
+                account: "".to_string(), // Default wallet account.
+                r#type: 4,               // Taproot address.
+                change: false,
+            })
+            .await
+        {
+            Ok(response) => Ok(GetFundingAddressResponse {
+                address: response.into_inner().addr,
+            }),
+            Err(e) => Err(LightningRpcError::FailedToGetFundingAddress {
+                failure_reason: format!("Failed to get funding address {e:?}"),
+            }),
+        }
+    }
+
+    async fn open_channel(
+        &self,
+        pubkey: String,
+        channel_size_sats: u64,
+        push_amount_sats: u64,
+    ) -> Result<EmptyResponse, LightningRpcError> {
+        let mut client = Self::connect(
+            self.address.clone(),
+            self.tls_cert.clone(),
+            self.macaroon.clone(),
+        )
+        .await?;
+
+        match client
+            .lightning()
+            .open_channel(OpenChannelRequest {
+                sat_per_vbyte: 0,
+                node_pubkey: hex::decode(pubkey).map_err(|e| {
+                    LightningRpcError::FailedToOpenChannel {
+                        failure_reason: format!("Failed to decode pubkey {e:?}"),
+                    }
+                })?,
+                node_pubkey_string: String::new(),
+                local_funding_amount: channel_size_sats.try_into().expect("u64 -> i64"),
+                push_sat: push_amount_sats.try_into().expect("u64 -> i64"),
+                target_conf: 0,
```

Comment:
Nit: I think you can use `..Default::default()` syntax if you want to make this a tad less verbose

Reply:
Oh snap, I wasn't familiar with this syntax. Much cleaner!